src/device/r4300/r4300_core.h

```diff
@@ -158,6 +167,25 @@ struct r4300_core
 #endif
 #endif
 
+    /* Memory accesses variables */
+    uint64_t* rdword;
+
+#if NEW_DYNAREC != NEW_DYNAREC_ARM
+/* ARM dynarec uses a different memory layout */
+    uint32_t wmask;
+
+    union {
+        uint32_t wword;
+        uint64_t wdword;
+    };
+
+    uint32_t address;
+#endif
+
+#if NEW_DYNAREC == NEW_DYNAREC_ARM
+    char extra_memory[33554432];
```

Comment:
We should take a look at the map file but ideally extra_memory should not be to far from the functions in linkage_arm.S to limit trampoline jumps.